grafter.repository/src/clj/grafter_2/rdf4j/repository.clj

```diff
@@ -137,20 +140,127 @@
   "Given a URL as a String return a Sesame HTTPRepository for e.g.
   interacting with the OpenRDF Workbench."
   [repo-url]
-  (doto (HTTPRepository. repo-url)
-    (.initialize)))
+  (HTTPRepository. repo-url))
+
+
+(defn make-http-client-builder
+  "Returns an unbuilt apache http-client configuration.
+
+  Takes a map of the following key value pairs:
+
+  - `:grafter.http/max-conn-total`               Total number of concurrent connections on this http client (default 10)
+  - `:grafter.http/max-conn-per-route`           Maximum number of concurrent connections allowed per endpoint (default 10)
+  - `:grafter.http/conn-time-to-live`            How long until an idle TCP connection in the pool will be closed (default 60)
+  - `:grafter.http/conn-time-to-live-timeunit`   Timeunits for the above (default TimeUnit/SECONDS)
+  - `:grafter.http/user-agent`                   The http user agent to report to the server (default \"grafter\")
+  "
+  [{:grafter.http/keys [max-conn-total
+                        max-conn-per-route
+                        conn-time-to-live
+                        conn-time-to-live-timeunit
+                        user-agent]
+    :or {max-conn-total 10
+         max-conn-per-route 10
+         conn-time-to-live 60
+         conn-time-to-live-timeunit TimeUnit/SECONDS
+         user-agent "grafter"
+         }}]
+  (-> (HttpClients/custom)
+      ;;(.useSystemProperties) ;; we could do this but better to be explict in clj
```

Comment:
Is it? I'd argue that having to follow the dependencies to this level (if you're in muttnik this is 2 libraries deep!) to find the defaults in the code is worse than assuming the client respects the system defaults.

Reply:
I ummed and ahed a little over this, but thought for now it was probably better to make this explicit.  

Firstly it's worth stating these properties don't really represent "System" defaults at all... they're not really things that are set universally at the O/S level.  They simply refer to arguments supplied to the JVM via system properties i.e. the `-Dsystem.property.key=value` mechanism.

It is the case that some of these properties are for setting things like http proxies and keystore things, but to my knowledge we've never needed to configure those to anything other than platform defaults (which we should also be getting here -- it's just you won't be able to reconfigure those things out of the box via system properties).

I should add that in the past we've had some problems because some of the properties that the apache http client uses also have a meaning in other parts of the JVM, which means you can't configure them independently.  I forget what the specific property was though...